mapbox_gl_platform_interface/lib/src/mapbox_gl_platform_interface.dart

```diff
@@ -210,4 +210,19 @@ abstract class MapboxGlPlatform {
     throw UnimplementedError(
         'setSymbolTextIgnorePlacement() has not been implemented.');
   }
+
+  Future<void> addSource(String sourceId, String geojson) async {
+    throw UnimplementedError(
+        'setSymbolTextIgnorePlacement() has not been implemented.');
+  }
+
+  Future<void> addSymbolLayer(String sourceId, String layerId, Map<String, String> properties) async {
```

Comment:
maybe rename this to `addSymbolLayerDynamic` so that we can add a typed version  called `addSymbolLayer`  in the future?